core/src/test/java/com/linecorp/armeria/client/endpoint/WeightRampingUpStrategyTest.java

```diff
@@ -0,0 +1,536 @@
+                                      .containsExactlyInAnyOrder(
+                                              Endpoint.of("foo.com"), Endpoint.of("foo1.com")
+                                      );
+        return selector;
+    }
+
+    private static List<Endpoint> endpointsFromSelectorEntry(RampingUpEndpointWeightSelector selector) {
+        final ImmutableList.Builder<Endpoint> builder = new ImmutableList.Builder<>();
+        final List<Entry> entries = selector.endpointSelector().entries();
+        entries.forEach(entry -> builder.add(entry.endpoint()));
+        return builder.build();
+    }
+
+    private void addSecondEndpoints(DynamicEndpointGroup endpointGroup,
+                                    RampingUpEndpointWeightSelector selector) {
+        endpointGroup.addEndpoint(Endpoint.of("bar.com"));
+        endpointGroup.addEndpoint(Endpoint.of("bar1.com"));
+
+        final Deque<EndpointsInUpdatingEntry> endpointsInUpdatingEntries = selector.endpointsInUpdatingEntries;
+        assertThat(endpointsInUpdatingEntries).hasSize(1);
+        final Set<EndpointAndStep> endpointAndSteps = endpointsInUpdatingEntries.peek().endpointAndSteps();
+        assertThat(endpointAndSteps).containsExactlyInAnyOrder(
+                endpointAndStep(Endpoint.of("bar.com"), 1, 100),
+                endpointAndStep(Endpoint.of("bar1.com"), 1, 100));
+        final List<Endpoint> endpointsFromEntry = endpointsFromSelectorEntry(selector);
+        assertThat(endpointsFromEntry).usingElementComparator(new EndpointComparator())
+                                      .containsExactlyInAnyOrder(
+                                              Endpoint.of("foo.com"), Endpoint.of("foo1.com"),
+                                              // 1000 * (1 / 10) => weight * (step / numberOfSteps)
+                                              Endpoint.of("bar.com").withWeight(100),
+                                              Endpoint.of("bar1.com").withWeight(100)
+                                      );
+    }
+
+    private static EndpointAndStep endpointAndStep(Endpoint endpoint, int step, int currentWeight) {
+        final EndpointAndStep endpointAndStep = new EndpointAndStep(endpoint, step);
+        endpointAndStep.currentWeight(currentWeight);
+        return endpointAndStep;
+    }
+
+    /**
+     * A Comparator which includes the weight of an endpoint to compare.
+     */
+    static class EndpointComparator implements Comparator<Endpoint>, Serializable {
+        private static final long serialVersionUID = -3534596922171048613L;
+
+        @Override
+        public int compare(Endpoint o1, Endpoint o2) {
+            if (o1.host().equals(o2.host()) &&
+                Objects.equals(o1.ipAddr(), o2.ipAddr()) &&
+                o1.weight() == o2.weight()) {
+                if (o1.hasPort() || o2.hasPort() && o1.port() == o2.port()) {
+                    return 0;
+                }
+                return 0;
```

Comment:
This `if` condition seems not to work properly. Regardless of it, this block will return `0`

Reply:
😱